File: .tmp-query2.py
import sqlite3, json
TEAM = 'team_tech-port-02-safety-license'
db = sqlite3.connect('db/nco.db')
db.row_factory = sqlite3.Row
rows = db.execute("""
SELECT id, metadata_json, error, status,
  LENGTH(COALESCE(response,'')) as response_len,
  COALESCE(heartbeat_seq,0) as heartbeat_seq
  FROM tasks WHERE team_id=? AND created_at >= datetime('now','-48 hours')
""", (TEAM,)).fetchall()

def parse(s):
    try:
        return json.loads(s)
    except Exception:
        return {}

retro = []
for r in rows:
    m = parse(r['metadata_json'])
    top = m.get('attemptedAgents') if isinstance(m.get('attemptedAgents'), list) else []
    hist = m.get('escalationHistory') if isinstance(m.get('escalationHistory'), list) else []
    maxH = []
    for h in hist:
        a = h.get('attemptedAgents') if isinstance(h, dict) and isinstance(h.get('attemptedAgents'), list) else []
        if len(a) > len(maxH):
            maxH = a
    if len(maxH) > 0 and len(top) < len(maxH):
        retro.append({
            'id': r['id'], 'status': r['status'], 'top': top, 'hist': maxH,
            'error': r['error'], 'response_len': r['response_len'], 'heartbeat_seq': r['heartbeat_seq']
        })
print('team02_retrograde_count=' + str(len(retro)))
print(json.dumps(retro, indent=2))
db.close()

File: .tmp-query2.mjs
import Database from 'better-sqlite3';
const db = new Database('db/nco.db',{readonly:true});
const TEAM='team_tech-port-02-safety-license';
const rows = db.prepare(`SELECT id, metadata_json, error, status,
  LENGTH(COALESCE(response,'')) as response_len,
  COALESCE(heartbeat_seq,0) as heartbeat_seq
  FROM tasks WHERE team_id=? AND created_at >= datetime('now','-48 hours')`).all(TEAM);
function parse(s){try{return JSON.parse(s)}catch{return{}}}
const retro=[];
for(const r of rows){
  const m=parse(r.metadata_json);
  const top=Array.isArray(m.attemptedAgents)?m.attemptedAgents:[];
  const hist=Array.isArray(m.escalationHistory)?m.escalationHistory:[];
  let maxH=[];
  for(const h of hist){const a=Array.isArray(h?.attemptedAgents)?h.attemptedAgents:[]; if(a.length>maxH.length)maxH=a;}
  if(maxH.length>0 && top.length<maxH.length){
    retro.push({id:r.id,status:r.status,top,hist:maxH,error:r.error,response_len:r.response_len,heartbeat_seq:r.heartbeat_seq});
  }
}
console.log('team02_retrograde_count='+retro.length);
console.log(JSON.stringify(retro,null,2));
db.close();
